File: 3/producto-frontend/src/services/CategoriaService.js
import axios from 'axios';
import CategoriaModel from '../model/CategoriaModel';

class CategoriaService {

    constructor() {
        this.urlServicio = 'http://localhost:8081/api/categoria';
        this.listaCategoria = [];
    }

    findAll = async () => {
        await axios.get(this.urlServicio)
            .then((res) => {
                res.data.forEach(c => {
                    if (c.estcat === 'A') {
                        this.listaCategoria.push(new CategoriaModel(c));
                    }
                });
                console.log(this.listaCategoria);

            })
            .catch(console.log);

        return this.listaCategoria;
    }


    findOne = (id) => {
        axios.get(this.urlServicio + '/' + id)
            .then((res) => {
                this.categoria = new CategoriaModel(res.data);
            })
            .catch(console.log);
        return this.categoria;
    }

    save = (categoria) => {
        axios.post(this.urlServicio, categoria)
            .then(console.log)
            .catch(console.log);
    }

    update = (categoria) => {
        axios.put(this.urlServicio, categoria)
            .then(console.log)
            .catch(console.log);
    }

    delete = (id) => {
        axios.delete(this.urlServicio + '/' + id)
            .then(console.log)
            .catch(console.log);
    }

}

export default CategoriaService;

File: 0/categoria_producto/src/main/java/com/martinsaman/categoria_producto/service/CategoriaService.java
package com.martinsaman.categoria_producto.service;

import com.martinsaman.categoria_producto.model.Categoria;
import com.martinsaman.categoria_producto.repository.ICategoria;
import com.martinsaman.categoria_producto.repository.IProducto;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;
import org.springframework.web.bind.annotation.RequestBody;

import java.util.List;
import java.util.Optional;

@Service
public class CategoriaService {

    @Autowired
    private ICategoria categoriaRepo;

    @Autowired
    private IProducto productoRepo;

    public ResponseEntity<List<Categoria>> findAll(){
        return new ResponseEntity<>(categoriaRepo.findAll(), HttpStatus.OK);
    }

    public ResponseEntity<Categoria> save(
            Categoria categoria
    ){
        return new ResponseEntity<>(categoriaRepo.save(categoria),HttpStatus.CREATED);
    }

    public ResponseEntity<Categoria> update(
            Categoria categoria
    ){
        Optional<Categoria> categoriaOptional = categoriaRepo.findById(categoria.getId());
        if (categoriaOptional.isPresent()){
            productoRepo.findAllByCategoria_Id(categoria.getId())
            .forEach((p)->{
                p.setCategoria(categoria);
                productoRepo.save(p);
            });
            return new ResponseEntity<>(categoriaRepo.save(categoria),HttpStatus.CREATED);
        }

        return new ResponseEntity<>(null, HttpStatus.NO_CONTENT);
    }

    public ResponseEntity<Void> delete(
            Categoria categoria
    ){
        categoriaRepo.delete(categoria);
        return new ResponseEntity<>(HttpStatus.CREATED);
    }


}
